Update `Variable.set` return value + docs (#13557)
PrefectHQ/prefect: docs/guides/variables.md

### Through the CLI

- `prefect variable set` creates or updates a variable.
- `prefect variable get` retrieves a variable's value.
- `prefect variable unset` deletes a variable.
- `prefect variable ls` lists all variables.

> - `prefect variable inspect` shows a variable's details.

Reviewer: What does "details" mean here?

- 
## Accessing variables

In addition to the UI and API, you can reference variables in code and in certain Prefect configuration files.

### In Python code

You can interact with Variables via the Python SDK utilizing the `get`, `set`, and `unset` methods. 

> Note that these methods are invoked either synchronously or asynchronously, depending on the context in which they are called

Reviewer 2: ```suggestion
Note that these methods are invoked either synchronously or asynchronously, depending on the context in which they are called. 
```

> while inside of a `flow` or `task`. Outside of a `flow` or `task`, these methods are always asynchronous.

Reviewer 2: ```suggestion
Outside of a `flow` or `task`, these methods are always asynchronous.
```

#### In an asynchronous context:
```python
from prefect import flow
from prefect.variables import Variable

@flow
async def my_async_flow():
    # set a variable
    answer = await Variable.set(name="the_answer", value="42")
    print(answer) # 42
    
    # get a variable
    answer = await Variable.get("the_answer")
    print(answer) # 42
    
    # overwrite an existing variable
    answer = await Variable.set(name="the_answer", value="43", overwrite=True)
    print(answer) # 43
    
    # unset a variable
    await Variable.unset("the_answer")
    
    # get a variable that doesn't exist
    answer = await Variable.get("not_the_answer")
    print(answer) # None
    
    # get a variable that doesn't exist with a default
    answer = await Variable.get("not_the_answer", default="42")
    print(answer) # 42
```

> While inside a `flow` or `task`, they should be invoked synchronously or asynchronously depending on if the current 

Reviewer 2: ```suggestion
While inside a `flow` or `task`, they should be invoked synchronously or asynchronously depending on whether the current context is synchronous or asynchronous.
```

> 

Reviewer 2: I think this is a good spot to note what's happening here with: "While inside of a `flow` or `task`...
